src/world.cpp

```diff
@@ -2208,7 +2208,7 @@ void World_GenRoom(struct room_s *room, class VT_Level *tr)
     room->alternate_room = NULL;
     room->base_room = NULL;
 
-    if((tr_room->alternate_room >= 0) && ((uint32_t)tr_room->alternate_room < tr->rooms_count))
+    if((tr_room->alternate_room >= 0) && ((uint32_t)tr_room->alternate_room < tr->rooms_count) && (room->id < tr_room->alternate_room))
```

Comment:
You are basically only swapping room->alternativeRoom, but you need to swap the rooms _and_ then the alternative IDs; as your pseudo-room only consists of the alternative room ID, you need to add another `swap(&rooms[room1], &rooms[room2])`. Maybe better use a struct with 2 members (an id for debugging the room data swap, and the alternative room id) in your case.